src/inmanta/data/model.py

```diff
@@ -749,14 +748,13 @@ class LoginReturn(BaseModel):
     user: User
 
 
-def is_resource_id(v: ResourceIdStr, info: pydantic.ValidationInfo) -> str:
-    try:
-        return resources.Id.parse_id(v).resource_str()
-    except Exception:
-        raise ValueError(f"Validation failed for {info.field_name}")
+def check_resource_id_str(v: str) -> ResourceIdStr:
```

Comment:
Tiny preference to make this "private"
```suggestion
def _check_resource_id_str(v: str) -> ResourceIdStr:
```

Not required.